File: gravitee-management-api-model/src/main/java/io/gravitee/management/model/alert/AlertTriggerEntity.java
/**
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.gravitee.management.model.alert;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.gravitee.alert.api.trigger.Trigger;

import java.util.Date;

/**
 * @author Azize ELAMRANI (azize.elamrani at graviteesource.com)
 * @author GraviteeSource Team
 */
public class AlertTriggerEntity extends Trigger {

    private String description;

    @JsonProperty("reference_type")
    private AlertReferenceType referenceType;

    @JsonProperty("reference_id")
    private String referenceId;

    @JsonProperty("created_at")
    private Date createdAt;

    @JsonProperty("updated_at")
    private Date updatedAt;

    private String type;

    protected AlertTriggerEntity(String id, String name, String source, Severity severity, boolean enabled) {
        super(id, name, severity, source, enabled);
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public AlertReferenceType getReferenceType() {
        return referenceType;
    }

    public void setReferenceType(AlertReferenceType referenceType) {
        this.referenceType = referenceType;
    }

    public String getReferenceId() {
        return referenceId;
    }

    public void setReferenceId(String referenceId) {
        this.referenceId = referenceId;
    }

    public Date getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Date createdAt) {
        this.createdAt = createdAt;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @Override
    public String toString() {
        return "AlertTriggerEntity{" +
                "description='" + description + '\'' +
                ", referenceType=" + referenceType +
                ", referenceId='" + referenceId + '\'' +
                ", createdAt=" + createdAt +
                ", updatedAt=" + updatedAt +
                '}';
    }
}

File: gravitee-apim-console-webui/src/entities/alerts/alertTriggerEntity.ts
/*
 * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Dampening } from './dampening';
import { AlertCondition } from './conditions';
import { NotificationPeriod } from './notificationPeriod';
import { AlertNotification } from './notification';

import { Scope } from '../alert';

export const ALERT_SEVERITIES = ['INFO', 'WARNING', 'CRITICAL'] as const;
export type AlertSeverity = (typeof ALERT_SEVERITIES)[number];

export interface AlertEventRuleEntity {
  event: string;
}

export interface AlertTriggerEntity {
  id: string;
  name: string;
  description: string;
  reference_type: Scope;
  reference_id: string;
  created_at: Date;
  updated_at: Date;
  type: string;
  last_alert_at: Date;
  last_alert_message: string;
  counters: Record<string, number>;
  template: boolean;
  event_rules: AlertEventRuleEntity[];
  parent_id: string;
  environment_id: string;
  severity: AlertSeverity;
  enabled: boolean;
  conditions?: AlertCondition[];
}

export interface NewAlertTriggerEntity extends Trigger {
  reference_type?: string;
  reference_id?: string;
  type?: string;
  template?: boolean;
  eventRules?: AlertEventRuleEntity[];
}

interface Trigger {
  id?: string;
  severity?: AlertSeverity;
  source?: string;
  name?: string;
  description?: string;
  conditions?: AlertCondition[];
  notifications?: AlertNotification[];
  dampening?: Dampening;
  metadata?: Record<string, Record<string, string>>;
  enabled?: boolean;
  filters?: AlertCondition[];
  notificationPeriods?: NotificationPeriod[];
}
